Mathlib/RingTheory/Radical.lean

```diff
@@ -168,3 +179,49 @@ theorem radical_neg {a : R} : radical (-a) = radical a :=
   radical_eq_of_associated Associated.rfl.neg_left
 
 end UniqueFactorizationDomain
+
+open UniqueFactorizationDomain
+namespace EuclideanDomain
+
+variable {E : Type*} [EuclideanDomain E] [NormalizationMonoid E] [UniqueFactorizationMonoid E]
+
+/-- Division of an element by its radical in an Euclidean domain. -/
+def divRadical (a : E) : E := a / radical a
+
+theorem radical_mul_divRadical (a : E) : radical a * divRadical a = a := by
+  rw [divRadical]
+  rw [← EuclideanDomain.mul_div_assoc]
+  ·refine mul_div_cancel_left₀ _ (radical_ne_zero a)
+  exact radical_dvd_self a
+
+theorem divRadical_mul_radical (a : E) : divRadical a * radical a = a := by
+  rw [mul_comm]
+  exact radical_mul_divRadical a
+
+theorem divRadical_ne_zero {a : E} (ha : a ≠ 0) : divRadical a ≠ 0 := by
+  rw [← radical_mul_divRadical a] at ha
+  exact right_ne_zero_of_mul ha
+
+theorem divRadical_isUnit {u : E} (hu : IsUnit u) : IsUnit (divRadical u) := by
+  rwa [divRadical, radical_unit_eq_one hu, EuclideanDomain.div_one]
+
+theorem eq_divRadical {a x : E} (h : radical a * x = a) : x = divRadical a := by
+  apply EuclideanDomain.eq_div_of_mul_eq_left (radical_ne_zero a)
+  rwa [mul_comm]
+
+theorem divRadical_mul {a b : E} (hab : IsCoprime a b) :
+    divRadical (a * b) = divRadical a * divRadical b := by
+  symm; apply eq_divRadical
+  rw [radical_mul hab]
+  rw [mul_mul_mul_comm, radical_mul_divRadical, radical_mul_divRadical]
+
+theorem divRadical_dvd_self (a : E) : divRadical a ∣ a := by
+  exact Dvd.intro (radical a) (divRadical_mul_radical a)
+
+protected theorem IsCoprime.divRadical {a b : E} (h : IsCoprime a b) :
```

Comment:
What about calling this `_root_.IsCoprime.divRadical`?

Reply:
Honestly, I don't know which one would be better (`protected` vs `_root_`), could you elaborate more?